torch/_dynamo/convert_frame.py

```diff
@@ -260,10 +260,17 @@ def _convert_frame_assert(
             recompiles_log.isEnabledFor(logging.DEBUG) or config.error_on_recompile
         ):
             if is_guard_failure_reporting_enabled():
-                message = (
-                    f"Recompiling function {code.co_name} in {code.co_filename}:{code.co_firstlineno}",
-                    f"triggered by the following guard failure: {str(guard_failures[code][-1])}",
-                )
+                if config.report_all_guard_failures:
+                    failures = str(guard_failures[code][-1]).strip().split("\n")
+                    message = (
+                        f"Recompiling function {code.co_name} in {code.co_filename}:{code.co_firstlineno}",
+                        f"triggered by the following guard failures: {failures}",
+                    )
+                else:
+                    message = (
+                        f"Recompiling function {code.co_name} in {code.co_filename}:{code.co_firstlineno}",
+                        f"triggered by the following guard failure: {str(guard_failures[code][-1])}",
+                    )
```

Comment:
Is the if branch just reporting the same thing than the second one but with a `.strip().split("\n")`?

Reply:
yes, I guess I can reduce code dup with "guard failure(s)"